spec/models/printables/group_spec.rb

```diff
@@ -82,75 +61,142 @@
+                  labels: [
                     { label: { barcode: '2', barcode2d: '2', top_line: '', middle_line: nil, bottom_line: '' } },
                     { label: { barcode: '1', barcode2d: '1', top_line: '', middle_line: nil, bottom_line: '' } }
                   ]
                 }
+              }.to_json,
+              headers: post_headers
+            )
+            .to_return(
+              status: 200,
+              body: '{"message":"labels successfully printed"}',
+              headers: {
+                'Content-Type' => 'application/json'
               }
-            }
-          }.to_json
-        )
-        .to_return(
-          status: 200,
-          body: '{ "message": "labels successfully printed" }',
-          headers: {
-            'Content-Type' => 'application/vnd.api+json'
-          }
-        )
-      group.print(config, 'user1')
-    end
+            )
+        group.print(config, 'user1')
+        expect(request).to have_been_made
+      end
 
-    context 'when an asset does not have barcode' do
-      let(:asset1) { create(:asset, facts: props1, barcode: nil) }
-      it 'does not print it' do
-        stub_request(:post, 'http://localhost:10000/v1/print_jobs')
-          .with(
-            body: {
-              data: {
-                type: 'print_jobs',
-                attributes: {
-                  printer_name: 'printer1',
-                  label_template_id: 1,
-                  labels: {
-                    body: [{ label: { barcode: '2', barcode2d: '2', top_line: '', middle_line: nil, bottom_line: '' } }]
+      context 'when an asset does not have barcode' do
+        let(:asset1) { create(:asset, facts: facts1, barcode: nil) }
+        xit 'does not print it' do
+          request =
+            stub_request(:post, "#{uri}/print_jobs")
+              .with(
+                body: {
+                  print_job: {
+                    printer_name: 'printer1',
+                    label_template_name: 'se_ean13_96tube',
+                    labels: [
+                      { label: { barcode: '2', barcode2d: '2', top_line: '', middle_line: nil, bottom_line: '' } }
+                    ]
                   }
+                }.to_json,
+                headers: post_headers
+              )
+              .to_return(
+                status: 200,
+                body: '{"message":"labels successfully printed"}',
+                headers: {
+                  'Content-Type' => 'application/json'
                 }
-              }
-            }.to_json
-          )
-          .to_return(
-            status: 200,
-            body: '{ "message": "labels successfully printed" }',
-            headers: {
-              'Content-Type' => 'application/vnd.api+json'
-            }
-          )
+              )
 
-        group.print(config, 'user1')
+          group.print(config, 'user1')
+          expect(request).to have_been_made
+        end
```

Comment:
Could also add a test when the response back from PMB isn't successful. Would need to confirm with PMB but the response body contains something like `data: { errors: [{detail: ..}]}` and something slightly odd like `{source: { pointer: ..}}`

Reply:
this is more relevant to test below `it 'handles failure with a custom exception'`